src/general-components/Export/JSONExporter.tsx

```diff
@@ -10,13 +10,38 @@ class JSONExporter<D> extends Exporter<D> {
         super("JSON", "json", "application/json");
     }
 
-    protected onExport = async (data: SaveResource<D>): Promise<BlobPart[]> => {
-        return [JSON.stringify(data.data, null, 4)];
+    protected onExport = async (data: SaveResource<D>, resources: ResourcesType): Promise<BlobPart[]> => {
+        let allResources = [];
+
+        // Add resources to export
+        for (let entry of Array.from(resources.entries())) {
+            let name = entry[0];
+            let value = entry[1];
+            allResources.push({
+                name: name,
+                file: await this.getDataURL(new Uint8Array(await value.file.arrayBuffer()), name, value.file.type),
+                type: value.file.type
+            });
+        }
+
+        return [JSON.stringify(Object.assign(Object.assign({}, data.data), {
```

Comment:
Object.assign accepts more than one argument as a result you could do this:
```suggestion
        return [JSON.stringify(Object.assign({}, data.data, {
```